crates/proof-of-sql/src/sql/proof/provable_query_result.rs

```diff
@@ -51,29 +43,29 @@ impl ProvableQueryResult {
     /// This function is available to allow for easy creation for testing.
     #[cfg(test)]
     #[must_use]
-    pub fn new_from_raw_data(num_columns: u64, indexes: Indexes, data: Vec<u8>) -> Self {
+    pub fn new_from_raw_data(num_columns: u64, table_length: u64, data: Vec<u8>) -> Self {
         Self {
             num_columns,
-            indexes,
+            table_length,
             data,
         }
     }
 
     /// Form intermediate query result from index rows and result columns
     #[must_use]
-    pub fn new<'a, S: Scalar>(indexes: &'a Indexes, columns: &'a [Column<'a, S>]) -> Self {
+    pub fn new<'a, S: Scalar>(table_length: u64, columns: &'a [Column<'a, S>]) -> Self {
```

Comment:
I'm suspicious here:
```suggestion
    pub fn new<'a, S: Scalar>(table_length: u64, columns: &'a [Column<'a, S>]) -> Self {
    assert!(columns.iter().all(|column| table_length <= column.len()))
```